### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by coskucinkilic**
# Motivation

Create tests that are missing to check feature flag and correct neuron id passed down

# Changes

NnsNeuronAdvancedSection.spec and po has been updated, 

# Tests

1. Test that the action is visible only if the feature flag is enabled.

2. Some way to be confident that the neuron passed to NnsNeuronAdvancedSection is the neuron passed to NnsNeuronPublicVisibilityAction.

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary

### CODE REVIEW

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts [hunk 1]
```diff
@@ -246,4 +248,36 @@ describe("NnsNeuronAdvancedSection", () => {
...
+  it("should pass the correct neuron to NnsNeuronPublicVisibilityAction", async () => {
+    overrideFeatureFlagsStore.setFlag("ENABLE_NEURON_VISIBILITY", true);
+
+    const po = renderComponent(createMockNeuron(123));
+
+    expect(await po.getNnsNeuronPublicVisibilityActionPo().isPresent()).toBe(
+      true
+    );
+
+    expect(await po.getNeuronId()).toBe("123");
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:281]: This neuron ID is not read from the `NnsNeuronPublicVisibilityAction` component, is it?

Comment (coskucinkilic) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:281]: `NnsNeuronPublicVisibilityAction ` doesn't have a neuron id, this is read from the list of key value pairs below the action.

Comment (dskloetd) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:281]: Would it be possible to pass a neuron with/without visibility and see that the text is different?
If not, then I guess we can add more coverage once the button is functional.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts (hunk 1 above)
Comment (coskucinkilic) [reply frontend/src/tests/lib/components/neuron-detail/NnsNeuronAdvancedSection.spec.ts:281]: The next pr for Modal has some tests related to change of text in NnsNeuronPublicVisibilityAction

**Pull request merged**